modules/openapi-generator/src/main/resources/dart2/api_helper.mustache

```diff
@@ -67,39 +67,27 @@ String parameterToString(dynamic value) {
 Future<String> _decodeBodyBytes(Response response) async {
   final contentType = response.headers['content-type'];
   return contentType != null && contentType.toLowerCase().startsWith('application/json')
-    ? response.bodyBytes == null ? null : utf8.decode(response.bodyBytes)
+    ? utf8.decode(response.bodyBytes)
     : response.body;
 }
 
-/// Returns a valid [T] value found at the specified Map [key], null otherwise.
-T mapValueOfType<T>(dynamic map, String key) {
-  final dynamic value = map is Map ? map[key] : null;
-  return value is T ? value : null;
-}
-
-/// Returns a valid Map<K, V> found at the specified Map [key], null otherwise.
-Map<K, V> mapCastOfType<K, V>(dynamic map, String key) {
-  final dynamic value = map is Map ? map[key] : null;
-  return value is Map ? value.cast<K, V>() : null;
-}
-
 /// Returns a valid [DateTime] found at the specified Map [key], null otherwise.
-DateTime mapDateTime(dynamic map, String key, [String pattern]) {
+DateTime mapDateTime(dynamic map, String key, [String? pattern]) {
   final dynamic value = map is Map ? map[key] : null;
   if (value != null) {
-    int millis;
+    int? millis;
     if (value is int) {
       millis = value;
     } else if (value is String) {
       if (pattern == _dateEpochMarker) {
         millis = int.tryParse(value);
       } else {
-        return DateTime.tryParse(value);
+        return DateTime.parse(value);
       }
     }
     if (millis != null) {
       return DateTime.fromMillisecondsSinceEpoch(millis, isUtc: true);
     }
   }
-  return null;
+  return DateTime(1970);
```

Comment:
this should probably still return null and callsites should be updated to check `required`/`nullable` and add `!` if necessary

Reply:
Thanks for spotting this, it was on my list, but I had to miss it, fixed.